studio-docs/source/schema/schema-reporting-protocol.mdx

```diff
@@ -1,85 +1,155 @@
+        Apollo->>Edge Server: ReportSchemaResponse(withCoreSchema = false, inSeconds = 62s)
         Note over Edge Server: Wait 62 seconds
-        Note over Edge Server: Go to top
-    else reportServerInfo request is malformed or otherwise invalid
-        Apollo->>Edge Server: Error(code, message)
+        Note over Edge Server: Go to Top
+    else reportSchema request is malformed or otherwise invalid
+        Apollo->>Edge Server: ReportSchemaError(code, message)
         Note over Edge Server: Stop until error is fixed
-    else reportServerInfo request failed with non-2xx HTTP response
+    else reportSchema request failed with non-2xx HTTP response
         Note over Edge Server: Wait 20 seconds
-        Note over Edge Server: Go to top
+        Note over Edge Server: Go to Top
     end
 ```
 
 ### Step-by-step description
 
-1. **On startup**, the edge server executes the `service.reportServerInfo` mutation, providing an `EdgeServerInfo` object with the server's details as input.
-    * This initial call **does not** include an `executableSchema` string.
-    * If this or any other `reportServerInfo` request fails with a non-2xx response, the edge server should retry after 20 seconds.
-2. **If the mutation succeeds**, Apollo responds with a `ReportServerInfoResponse` object. This response tells the edge server:
-    * How many seconds to wait before sending the next `reportServerInfo` request
-    * Whether the next `reportServerInfo` request should include the `executableSchema` that corresponds to the `executableSchemaId` provided in the previous request.
+1. **On startup**, the edge server executes the `reportSchema` mutation, providing a `SchemaReport` object with the server's details as input.
+    * See the `SchemaReport` object's [required fields](#required-fields).
+    * In this initial call, set the `coreSchema` argument of `reportSchema` to `null`.
+    * If this or any other `reportSchema` request fails with a non-2xx response, the edge server should retry after 20 seconds.
+2. **If the mutation succeeds**, Apollo responds with a `ReportSchemaResponse` object. This response tells the edge server:
+    * How many seconds to wait before sending the next `reportSchema` request (`inSeconds`)
+    * Whether the next `reportSchema` request should include the `coreSchema` that corresponds to the `coreSchemaHash` provided in the previous request (`withCoreSchema`)
 
-    **If the mutation fails**, Apollo responds with a `ReportServerInfoError` object. **In this case, the edge server should stop reporting its schema.**
+    **If the mutation fails**, Apollo responds with a `ReportSchemaError` object. **In this case, the edge server should stop reporting its schema.**
     * The `message` field of `ReportServerInfoError` provides a human-readable message describing the error.
     * Correct the error and deploy a new version of your edge server to resume schema reporting.
```

Comment:
I think we should call out that the error should _not_ cause the server to crash.